### PR TITLE
Add buffering vignette (fixes #46)

### LINKED ISSUE #46: Add a "sharp edges" FAQ
## Feature

It'd be useful to highlight potential problems users might have with spatialsample, and possible workarounds. This issue should collect any "sharp edges" of the package that we know about (but aren't planning on buffing out pre-release).

### PR DESCRIPTION
**Pull request by mikemahoney218**
This PR adds a vignette covering the "sharp edges" around buffering. It also adds the centroid clarification from #46 to the blocking documentation.

This PR fixes #46 .

### CODE REVIEW

**Comment by mikemahoney218** [comment]
This is going to fail because `main` is failing; fixes for that are in #59

**Comment by mikemahoney218** [comment]
The original reason was to debug _something_ to do with Mac CI, and then also to make building faster -- I'll try reverting and see if CI passes.

**Comment by mikemahoney218** [comment]
:sob: https://github.com/tidymodels/spatialsample/runs/6871375630?check_suite_focus=true#step:7:120

**Comment by mikemahoney218** [comment]
A re-run seems to have worked! Wish I knew the root cause though

**Review by hfrick — APPROVED** [review]
Looks great and illustrates the little gotchas well 👍 

It took fairly long to render for me. Should we think about if we want to turn any of the animations into a static plot?

**Comment by mikemahoney218** [comment]
I'm not 100% sure if profvis would actually catch the amount of time spent on gifski, but when calling `profvis::provis(rmarkdown::render)` most of the time is spent in the buffered vfold of the last chunk:

Screenshot from 2022-06-14 08-07-05

This tracks with my experience with this function -- buffering takes a _long_ time (it's most of our test suite runtime) and doing it twice for every point in the data set also takes a minute. That might be its own thing to work on (we might be able to optimize by calling `sf::st_distance()` once and indexing into that, rather than calling `sf::st_is_within_distance()` a bunch of times).

File: vignettes/buffering_for_spatial_cross_validation.Rmd [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+spatialsample aims to help you more accurately assess your models by introducing some distance between your analysis and assessment data. Spatially separating the data a model is trained on from the data it is evaluated against can provide a more realistic view of how well a given model will perform when extrapolating to new locations.
+
+One of the most common methods for introducing this distance is to "buffer" the assessment set, removing all points within a given distance from the analysis set to enforce a minimum space between data sets. This vignette walks through how to buffer your assessment folds in spatialsample, and some considerations about how those buffers are calculated.
+
+To begin, we're going to need to load spatialsample:
+
+```{r setup}
+library(spatialsample)
+```
```
[the remark this answers is not in the document]
Comment (mikemahoney218) [reply vignettes/buffering_for_spatial_cross_validation.Rmd:26]: I _believe_ we do -- examples from rsample:
https://rsample.tidymodels.org/articles/Working_with_rsets.html
https://rsample.tidymodels.org/articles/Applications/Survival_Analysis.html

examples from parsnip:
https://parsnip.tidymodels.org/articles/Submodels.html

**Comment by mikemahoney218** [comment]
re: the above, #65 should help somewhat

**Comment by mikemahoney218** [comment]
Yeah, after merging #65 to my local branch this vignette takes a bit less than half the time to render (2.2x speedup, if my math is right) and _now_ it seems like the GIFs are the major holdup:

Screenshot from 2022-06-14 09-42-45

**Comment by mikemahoney218** [comment]
extremely interested in the buildtime of b2cf61c :eyes:

**Comment by mikemahoney218** [comment]
12m pkgdown build https://github.com/tidymodels/spatialsample/runs/6885188727?check_suite_focus=true

Compared to 30m previously https://github.com/tidymodels/spatialsample/actions/runs/2491833820

Wow!

**Review by juliasilge — APPROVED** [review]
This is a really fantastic vignette 🙌 

My one question before we merge is if there is a better link to use for leave-disc-out CV than the CRAN page for sperrorest. A paper? The [pkgdown page from sperrorest](https://giscience-fsu.github.io/sperrorest/reference/partition_disc.html)?

**Comment by mikemahoney218** [comment]
sperrorest links to https://nhess.copernicus.org/articles/5/853/2005/nhess-5-853-2005.pdf for this function, which is a paper by one of the same authors as the package. They describe their methods on page 6; it reads to _me_ like they used `buffer` but not `radius` (so performed a buffered leave-some-out). I personally don't know a better academic reference. 

I'd be happy to link the pkgdown page. One concern: as we don't do the _exact_ same thing (sperrorest buffers a doughnut around a radius disc, while we buffer a distance around the test set), would it be confusing to directly link an alternative implementation?

**Comment by juliasilge** [comment]
Yeah, maybe so. What do you think about no link? I think the figures + text make it pretty clear what is happening.

**Comment by mikemahoney218** [comment]
Sounds good to me!

**Pull request merged**